Patch description:
=================
1) Add support for IPv6 logo test cases for multicast.


diff -Naur linux.orig/net/ipv6/ip6mr.c linux/net/ipv6/ip6mr.c
--- linux.orig/net/ipv6/ip6mr.c	2011-02-25 15:24:56.000000000 +0530
+++ linux/net/ipv6/ip6mr.c	2011-02-25 17:43:01.000000000 +0530
@@ -1504,10 +1504,23 @@
 	struct net_device *dev;
 	struct dst_entry *dst;
 	struct flowi fl;
+	struct net_device * input_dev = skb_dst(skb)->dev;
+	struct net_device * output_dev = vif->dev;
 
 	if (vif->dev == NULL)
 		goto out_free;
 
+	if (skb->len > output_dev->mtu ) {
+		printk ("ip6mr_queue_xmit stage2.\n");
+		skb->dev = input_dev;
+		icmpv6_send(skb, ICMPV6_PKT_TOOBIG, 0, output_dev->mtu, input_dev);
+		IP6_INC_STATS_BH(dev_net(skb_dst(skb)->dev), ip6_dst_idev(skb_dst(skb)), 
+		                 IPSTATS_MIB_INTOOBIGERRORS);
+		IP6_INC_STATS_BH(dev_net(skb_dst(skb)->dev), ip6_dst_idev(skb_dst(skb)), 
+		                 IPSTATS_MIB_FRAGFAILS);
+		goto out_free;
+	}
+
 #ifdef CONFIG_IPV6_PIMSM_V2
 	if (vif->flags & MIFF_REGISTER) {
 		vif->pkt_out++;
